R/getNeatOrder.R

```diff
@@ -0,0 +1,179 @@
+#' Sorting by radial theta angle
+#' 
+#' @description \code{getNeatOrder} sorts already ordinated data by the radial theta angle.
+#' This method is useful for organizing data points based on their angular 
+#' position in a 2D space, typically after an ordination technique such as PCA or NMDS 
+#' has been applied. 
+#' 
+#' The function takes in a matrix of ordinated data, optionally 
+#' centers the data using specified methods (mean, median, or none), and then calculates 
+#' the angle (theta) for each point relative to the centroid. The data points are then 
+#' sorted based on these theta values in either ascending or descending order. 
+#' 
+#' One significant application of this sorting method is in plotting heatmaps. 
+#' By using radial theta sorting, the relationships between data points can be preserved 
+#' according to the ordination method's spatial configuration, rather than relying on 
+#' hierarchical clustering, which may distort these relationships. This approach 
+#' allows for a more faithful representation of the data's intrinsic structure as captured 
+#' by the ordination process.
+#' 
+#' @param x A matrix containing the ordinated data to be sorted. Columns should represent the principal components (PCs) and rows should represent the entities being analyzed (e.g., features or samples).
+#' @param subset A vector specifying a subset of rows to be used and retained. If NULL, all rows are used.
+#' @param dimensions A vector of length 2 specifying the columns of the matrix to use for the X and Y coordinates.
+#' @param centering_method A character string specifying the method to center the data. Options are "mean", "median", or "none" if your data is already centred.
+#' @param decreasing A boolean that when true sorts the rows in a descending order by radial theta angle. Default is False.
+#' @param ... Additional arguments passed to other methods.
+#' @return A vector of row names in the sorted order.
+#' 
+#' @name getNeatOrder
+#' 
+#' @examples
+#' # Load required libraries
+#' library(mia)
+#' 
+#' # Load the dataset
+#' data(peerj13075)
+#' 
+#' # Agglomerate by Order and transform the data
+#' tse_order <- agglomerateByRank(peerj13075, rank = "order", onRankOnly = TRUE)
+#' tse_order <- transformAssay(tse_order, assay.type = "counts", method="relabundance", MARGIN = "samples", name="relabundance")
+#' tse_order <- transformAssay(tse_order, assay.type="relabundance", method="z", MARGIN = "features", name="z")
+#' z_transformed_data <- assay(tse_order, "z")
+#' 
+#' # Get the top taxa and perform PCA
+#' top_taxa <- getTopFeatures(tse_order, top = 10, assay.type="z")
```

Comment:
I think this is better to remove. Usually PCA would be done with all features in this context unless there are scalability issues. Agglomerating the data to the order level should already take care of that (dimensionality should be relatively low for fast calculation).

Reply:
Note that the name "z" was just changed to "scale" in the new mia version:
https://github.com/microbiome/mia/issues/583

-> Update this here as well?

Reply:
@antagomir which mia version is this? I'm on 1.12.0 and i can't seem to update it from that version, and scale isn't recognized on this version. Neither is margin="cols"

Reply:
very recent mia PR that was merged
https://github.com/microbiome/mia/pull/584

-> in fact I didn't check if they updated the version number.

OK the margin you can keep as is. I think that wasn't merged yet.

Reply:
Ah ok yeah, I can add in scale, although it's not quite working right now. Also, is this suitable in terms of the heatmap my example creates?

<img width="842" alt="Screenshot 2024-06-20 at 12 32 48" src="https://github.com/microbiome/miaViz/assets/116303632/c68a0193-c244-4be3-a7b1-482403fb6c9f">

Reply:
The samples have the ordering from getNeatOrder applied to them. Will wrap this up and the OMA chapter today btw so I can move onto the EBI stuff

Reply:
Ok this looks better. You could check the following however:

1) confirm manually that the rows (taxa) have zero mean, unit variance (they should, after "scale")
2) you can consider capping the colors at for instance to [-5, 5] (values exceeding these would be capped to -5 and 5, respectively; this is a common visualization practice and ok when transparently reported
3) I realize that the starting sample can be decided arbitrarily, there is no way to determine the "correct" first sample; you could test how changing the start sample (left most sample) would impact the figure. Here, the left and right ends would form a continuum as well (as the angle rotates full circle)

Reply:
ok thanks will do. yes all the data has near 0 mean and stdev of 1